plugins/redfish/fu-plugin-redfish.c

```diff
@@ -8,69 +8,179 @@
 
 #include <fwupdplugin.h>
 
-#include "fu-redfish-client.h"
+#include "fu-redfish-backend.h"
 #include "fu-redfish-common.h"
+#include "fu-redfish-network.h"
 #include "fu-redfish-smbios.h"
 
 struct FuPluginData {
-	FuRedfishClient		*client;
+	FuRedfishBackend	*backend;
 };
 
-gboolean
-fu_plugin_update (FuPlugin *plugin,
-		  FuDevice *device,
-		  GBytes *blob_fw,
-		  FwupdInstallFlags flags,
-		  GError **error)
-{
-	FuPluginData *data = fu_plugin_get_data (plugin);
-
-	return fu_redfish_client_update (data->client, device, blob_fw, error);
-}
-
 gboolean
 fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 {
 	FuPluginData *data = fu_plugin_get_data (plugin);
-	GPtrArray *devices;
+	g_autoptr(GPtrArray) devices = NULL;
+	g_autoptr(GError) error_local = NULL;
 
 	/* get the list of devices */
-	if (!fu_redfish_client_coldplug (data->client, error))
+	if (!fu_backend_coldplug (FU_BACKEND (data->backend), &error_local)) {
+		if (g_error_matches (error_local, FWUPD_ERROR, FWUPD_ERROR_AUTH_FAILED))
+			fu_plugin_add_flag (plugin, FWUPD_PLUGIN_FLAG_UNLOCK_REQUIRED);
```

Comment:
you don't want to fail coldplug though in this instance do you?  Don't you want to allow an unlock?

Reply:
I think I chose the wrong plugin flag. What I really wanted was something like FWUPD_PLUGIN_FLAG_AUTH_REQUIRED. The logic then that "fwupdmgr get-plugins" would clearly show why it's failed to coldplug with some nice red text.

Reply:
Fixed.